core/src/main/scala/kafka/coordinator/transaction/TransactionMarkerChannel.scala

```diff
@@ -54,41 +87,28 @@ class TransactionMarkerChannel(interBrokerListenerName: ListenerName,
 
   private[transaction]
   def drainQueuedTransactionMarkers(txnMarkerPurgatory: DelayedOperationPurgatory[DelayedTxnMarker]): Iterable[RequestAndCompletionHandler] = {
-    brokerStateMap.flatMap {case (brokerId: Int, destAndMarkerQueue: DestinationBrokerAndQueuedMarkers) =>
-      val markersToSend: java.util.List[CoordinatorEpochAndMarkers] = new util.ArrayList[CoordinatorEpochAndMarkers] ()
-      destAndMarkerQueue.markersQueue.drainTo (markersToSend)
-      markersToSend.groupBy{ epochAndMarker => (epochAndMarker.metadataPartition, epochAndMarker.coordinatorEpoch) }
-        .map { case((metadataPartition: Int, coordinatorEpoch:Int), buffer: mutable.Buffer[CoordinatorEpochAndMarkers]) =>
-          val txnMarkerEntries = buffer.flatMap{_.txnMarkerEntries }.asJava
-          val requestCompletionHandler = new TransactionMarkerRequestCompletionHandler(
-            this,
-            txnMarkerPurgatory,
-            CoordinatorEpochAndMarkers(metadataPartition, coordinatorEpoch, txnMarkerEntries),
-            brokerId)
-          RequestAndCompletionHandler(destAndMarkerQueue.destBrokerNode, new WriteTxnMarkersRequest.Builder(coordinatorEpoch, txnMarkerEntries), requestCompletionHandler)
-        }
+    brokerStateMap.flatMap {case (brokerId: Int, brokerRequestQueue: BrokerRequestQueue) =>
```

Comment:
@guozhangwang i wonder if before this we should do:
````
brokerStateMap.filter {case (_, brokerRequestQueue:BrokerRequestQueue) => 
   networkClient.isReady(brokerRequestQueue.node(), time)
```
So we don't bother queueing requests for a node that isn't ready? I was going to add this, but wasn't sure if there would be any gotchas, i.e., what if the node isn't ready as it has gone offline? Then these requests would never be queued